pkg/process/checks/net_test.go

```diff
@@ -145,3 +136,35 @@ func TestNetworkConnectionBatchingWithDNS(t *testing.T) {
 	}
 	assert.Equal(t, 4, total)
 }
+
+func TestBatchSimilarConnectionsTogether(t *testing.T) {
+	p := makeConnections(6)
+
+	p[0].Raddr.Ip = "1.1.2.3"
+	p[1].Raddr.Ip = "1.2.3.4"
+	p[2].Raddr.Ip = "1.3.4.5"
+	p[3].Raddr.Ip = "1.1.2.3"
+	p[4].Raddr.Ip = "1.2.3.4"
+	p[5].Raddr.Ip = "1.3.4.5"
+
+	cfg := config.NewDefaultAgentConfig(false)
+	cfg.MaxConnsPerMessage = 2
+
+	chunks := batchConnections(cfg, 0, p, map[string]*model.DNSEntry{}, "nid", nil)
+
+	assert.Len(t, chunks, 3)
+	total := 0
+	for _, c := range chunks {
+		connections := c.(*model.CollectorConnections)
+		total += len(connections.Connections)
+		assert.Equal(t, int32(3), connections.GroupSize)
+		assert.Equal(t, 2, len(connections.Connections))
+
+		// make sure the connections with similar remote addresses were grouped together
+		rAddr := connections.Connections[0].Raddr.Ip
+		for _, cc := range connections.Connections {
+			assert.Equal(t, rAddr, cc.Raddr.Ip)
+		}
+	}
+	assert.Equal(t, 6, total)
```

Comment:
Can we also have a test for the case when remote addresses are same but PID as different?